### DISCUSSION TITLE
【更灵活的广告嵌入】在文章指定的段落嵌入广告，How add Adsense in Heo Theme.

### DISCUSSION THREAD

**Discussion by Nexus-Gits** [tangly1024/NotionNext#2127] [Ideas]
Please tell me how add adsense code in heo site.

**Comment by tangly1024** [comment]
Hello! Sure, I'd be happy to help you add AdSense to the Heo theme. There are two main methods to do this:


1. Automatic Ads (Simpler):
    - Configure the ADSENSE_GOOGLE_ID in NotionNext.
    - Set up automatic ads, like banner and interstitial ads, in the Google AdSense backend.
    - Google AdSense will then automatically insert ads into your Heo site based on the configured settings.

    ![image](https://github.com/tangly1024/NotionNext/assets/15920488/aa256b54-3708-4911-817a-97ffabe4d135)


2. Custom Ad Units (More Complex):
    - Create four types of ad units in Google AdSense.
    - Configure their unit IDs and the ADSENSE_GOOGLE_ID in NotionNext. (Refer to [this link](https://github.com/tangly1024/NotionNext/blob/82dea161a0690d8360d21dc1aa8e64f86b0535b0/blog.config.js#L350C3-L355))

    ![image](https://github.com/tangly1024/NotionNext/assets/15920488/e3597f5b-9e90-4607-b75b-6016a2bddb1f)

3. After the configuration, use the AdSlot component ([AdSlot](https://github.com/tangly1024/NotionNext/blob/82dea161a0690d8360d21dc1aa8e64f86b0535b0/components/GoogleAdsense.js#L46)) in your Heo theme wherever you want to display ads.

    you can refer to the approach used in other themes, such as the Fukasawa theme. For instance, in the themes/fukasawa/index.js file, a similar method is employed:

```
  import { AdSlot } from '@/components/GoogleAdsense'
  
  const LayoutBase = ...{
                 {/*  Ad unit displayed here */}
  
                <div className='mt-2'>
                  <AdSlot type='native' />
                </div>
  }

```

This demonstrates how to integrate the AdSlot component in the Fukasawa theme. Feel free to adapt this example to add AdSense to your Heo theme. If you have any more questions or need further assistance, please let me know!

**Comment by Nexus-Gits** [comment]
How i set up, ads insert after every certain word later on every blog post.
